tests/bwrite.rs

```diff
@@ -17,128 +17,554 @@
-          "Price": 2,
-          "pk": "ichi",
-          "Details": {
-            "Age": 35,
-            "Misc": {
-              "dream": [
-                35,
-                null
-              ],
-              "hope": true
-            },
-            "Name": "Joe"
-          },
-          "ISBN": "111-1111111111"
-        }
-    */
+    let scan_cmd = c.args(&[
+        "--region",
+        "local",
+        "--table",
+        &table_name,
+        "scan",
+        "-o",
+        "json",
+    ]);
+    scan_cmd.assert().success().stdout(
+        predicate::str::is_match(r#"pk": "ni""#)?.and(predicate::str::is_match(r#"pk": "san""#)?),
+    );
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_batch_write_json_put_delete_multiple_tables() -> Result<(), Box<dyn std::error::Error>>
+{
+    let mut tm = util::setup().await?;
+    let table_name = tm
+        .create_temporary_table_with_items(
+            "pk",
+            None,
+            vec![util::TemporaryItem::new(
+                "ichi",
+                None,
+                Some(r#"{"null-field": null}"#),
+            )],
+        )
+        .await?;
+
+    let table_name2 = tm
+        .create_temporary_table_with_items(
+            "pk",
+            None,
+            vec![util::TemporaryItem::new(
+                "ichi",
+                None,
+                Some(r#"{"null-field": null}"#),
+            )],
+        )
+        .await?;
+
+    let tmpdir = Builder::new().tempdir()?; // defining stand alone variable here as tempfile::tempdir creates directory and deletes it when the destructor is run.
+    let batch_input_file_path = create_test_json_file(
+        "tests/resources/test_batch_write_put_delete_multiple_tables.json",
+        vec![&table_name, &table_name2],
+        &tmpdir,
+    );
+
     let mut c = tm.command()?;
-    let get_cmd = c.args(&["--region", "local", "--table", &table_name, "get", "ichi"]);
-    let output = get_cmd.output()?.stdout;
+    c.args(&[
+        "--region",
+        "local",
+        "bwrite",
+        "--input",
+        &batch_input_file_path,
+    ])
+    .output()?;
 
-    // more verification would be nice
-    assert_eq!(
-        true,
-        predicate::str::is_match("\"Dimensions\":")?.eval(String::from_utf8(output)?.as_str())
+    let mut c = tm.command()?;
+    let scan_cmd = c.args(&[
+        "--region",
+        "local",
+        "--table",
+        &table_name,
+        "scan",
+        "-o",
+        "json",
+    ]);
+    scan_cmd.assert().success().stdout(
+        predicate::str::is_match(r#"pk": "ni""#)?.and(predicate::str::is_match(r#"pk": "san""#)?),
+    );
+
+    let mut c = tm.command()?;
+    let scan_cmd = c.args(&[
+        "--region",
+        "local",
+        "--table",
+        &table_name2,
+        "scan",
+        "-o",
+        "json",
+    ]);
+    scan_cmd.assert().success().stdout(
+        predicate::str::is_match(r#"pk": "ni""#)?.and(predicate::str::is_match(r#"pk": "san""#)?),
     );
```

Comment:
Can we simplify this assertion using a loop?

Reply:
Fixed.